File: blog/src/main/kotlin/com/liux/blog/bean/po/Article.kt
package com.liux.blog.bean.po

import com.liux.blog.bean.NoArgsConstructor
import java.io.Serializable
import java.util.*

// 文章数据
@NoArgsConstructor
data class Article(
    // ID
    var id: Int? = null,
    // 标题
    var title: String? = null,
    // 内容
    var content: String? = null,
    // 分类ID
    var categoryId: Int? = null,
    // 作者ID
    var authorId: Int? = null,
    // 自定义url
    var url: String? = null,
    // 权重
    var weight: Int? = null,
    // 阅读量
    var views: Int? = null,
    // 允许评论
    var enableComment: Boolean? = null,
    // 状态
    var status: Int? = null,
    // 创建时间
    var createTime: Date? = null,
    // 更新时间
    var updateTime: Date? = null,

    /* 分类 */
    var category: Category? = null,
    /* 作者 */
    var author: User? = null,
    /* 标签 */
    var tags: List<Tag>? = null,
) : Serializable

File: blog/bean/api_vo/article.go
package api_vo

import (
	"blog/bean/po"
	"time"
)

type ArticleVO struct {
	Id            int        `json:"id"`
	Title         string     `json:"title"`
	CategoryId    int        `json:"categoryId"`
	Url           string     `json:"url"`
	Tags          []string   `json:"tags"`
	Content       string     `json:"content"`
	Weight        int        `json:"weight"`
	EnableComment bool       `json:"enableComment"`
	Status        int        `json:"status"`
	CreateTime    time.Time  `json:"createTime"`
	UpdateTime    *time.Time `json:"updateTime"`
}

func (a *ArticleVO) From(article po.Article) {
	a.Id = article.Id
	a.Title = article.Title
	a.CategoryId = article.CategoryId
	a.Url = article.GetSafeUrl()
	a.Tags = article.GetSafeTags()
	a.Content = article.Content
	a.Weight = article.Weight
	a.EnableComment = article.CommentStatus == po.ARTICLE_COMMENT_ENABLE
	a.Status = article.Status
	a.CreateTime = article.CreateTime
	a.UpdateTime = article.UpdateTime
}
